### PR TITLE
Change default log level to WARN

### PR DESCRIPTION
**Pull request by bmuschko**
### Context

See https://github.com/gradle/gradle/issues/1610 for more information (especially the open issues section). [Stage 4 passed on CI coverage.](https://builds.gradle.org/viewType.html?buildTypeId=Gradle_Branches_Checkpoints_Stage4FullCoverage&branch_Gradle_Branches_Checkpoints=bm-warn-default-log-level&tab=buildTypeStatusDiv)

Documentation will still need to be added. Maybe we can do it altogether with the overarching goal of grouping tasks.

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation including proper use of `@since` and `@Incubating` annotations for all public APIs
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by bmuschko** [comment]
@adammurdoch Would you mind having a look over this PR as well?

File: subprojects/core/src/main/java/org/gradle/caching/internal/BuildCacheServiceProvider.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public BuildCacheService createBuildCacheService() {
         } else if (canUseRemoteBuildCache && !startParameter.isOffline()) {
             buildCacheService = createStandaloneRemoteBuildService(remote);
         } else {
-            LOGGER.warn("Task output caching is enabled, but no build caches are configured or enabled.");
+            LOGGER.lifecycle("Task output caching is enabled, but no build caches are configured or enabled.");
```
Comment (eriwen) [inline subprojects/core/src/main/java/org/gradle/caching/internal/BuildCacheServiceProvider.java]: :o: Nitpick: Weak opinion that we keep this at warn.

File: subprojects/testing-base/src/test/groovy/org/gradle/api/internal/tasks/testing/logging/DefaultTestLoggingContainerTest.groovy [hunk 2]
```diff
@@ -48,7 +48,13 @@ class DefaultTestLoggingContainerTest extends Specification {
         def logging = container.get(LogLevel.WARN)
 
         expect:
-        hasUnchangedDefaults(logging)
+        logging.events == [TestLogEvent.FAILED] as Set
```
Comment (eriwen) [inline subprojects/testing-base/src/test/groovy/org/gradle/api/internal/tasks/testing/logging/DefaultTestLoggingContainerTest.groovy]: :o: Nitpick. We probably could reduce some duplication with these assertions and others in this class.

File: subprojects/core/src/main/java/org/gradle/caching/internal/BuildCacheServiceProvider.java [hunk 3]
```diff
@@ -64,7 +64,7 @@ public BuildCacheService createBuildCacheService() {
         boolean canUseRemoteBuildCache = remote != null && remote.isEnabled();
 
         if (canUseRemoteBuildCache && startParameter.isOffline()) {
-            LOGGER.warn("Remote build cache is disabled when running with --offline.");
+            LOGGER.lifecycle("Remote build cache is disabled when running with --offline.");
```
Comment (bmuschko) [inline subprojects/core/src/main/java/org/gradle/caching/internal/BuildCacheServiceProvider.java]: That's what you proposed here: https://github.com/gradle/gradle/pull/1952#commitcomment-21989913

File: subprojects/core/src/main/java/org/gradle/caching/internal/BuildCacheServiceProvider.java (hunk 1 above)
Comment (bmuschko) [inline subprojects/core/src/main/java/org/gradle/caching/internal/BuildCacheServiceProvider.java]: See comment above

File: subprojects/base-services/src/main/java/org/gradle/util/GradleVersion.java [hunk 4]
```diff
@@ -308,4 +308,17 @@ public int compareTo(Stage other) {
             return 0;
         }
     }
+
+    public boolean isSameOrNewer(String otherVersion) {
+        return isVersion(otherVersion) || this.compareTo(GradleVersion.version(otherVersion)) > 0;
+    }
+
+    public boolean isSameOrOlder(String otherVersion) {
```
Comment (bmuschko) [inline subprojects/base-services/src/main/java/org/gradle/util/GradleVersion.java]: This was actually coming from `DefaultGradleDistribution` but was moved. I didn't want to change too much of the testing infrastructure. It's now refactored and in use.

File: subprojects/testing-base/src/test/groovy/org/gradle/api/internal/tasks/testing/logging/DefaultTestLoggingContainerTest.groovy (hunk 2 above)
Comment (bmuschko) [inline subprojects/testing-base/src/test/groovy/org/gradle/api/internal/tasks/testing/logging/DefaultTestLoggingContainerTest.groovy]: done

**Comment by bmuschko** [comment]
> Would we want to change `OutputEventListenerBackedLoggerContext.DEFAULT_LOG_LEVEL`? What about the log level of the progress events?

Wouldn't that mean that we'd see all the tasks + their statuses in the output again? Isn't that what we want to avoid? We probably should _just_ change the log level in your PR that renders the task grouping header to `WARN` if I am not mistaken?

File: subprojects/base-services/src/main/java/org/gradle/util/GradleVersion.java (hunk 4 above)
Comment (eriwen) [inline subprojects/base-services/src/main/java/org/gradle/util/GradleVersion.java]: :x: Probably need `@since` here.

Comment (bmuschko) [inline subprojects/base-services/src/main/java/org/gradle/util/GradleVersion.java]: It's an internal class so no need.

**Pull request merged**